core/src/main/java/org/wildfly/extras/creaper/core/online/HttpModelControllerClient.java

```diff
@@ -0,0 +1,228 @@
+
+
+import org.apache.http.HeaderElement;
+import org.apache.http.HttpHeaders;
+import org.apache.http.HttpStatus;
+import org.apache.http.auth.AuthScope;
+import org.apache.http.auth.UsernamePasswordCredentials;
+import org.apache.http.client.config.AuthSchemes;
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.config.Registry;
+import org.apache.http.config.RegistryBuilder;
+import org.apache.http.conn.socket.ConnectionSocketFactory;
+import org.apache.http.conn.socket.PlainConnectionSocketFactory;
+import org.apache.http.conn.ssl.NoopHostnameVerifier;
+import org.apache.http.conn.ssl.SSLConnectionSocketFactory;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.BasicCredentialsProvider;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.impl.client.StandardHttpRequestRetryHandler;
+import org.apache.http.impl.conn.PoolingHttpClientConnectionManager;
+import org.apache.http.util.EntityUtils;
+import org.jboss.as.controller.client.ModelControllerClient;
+import org.jboss.as.controller.client.Operation;
+import org.jboss.as.controller.client.OperationMessageHandler;
+import org.jboss.as.controller.client.OperationResponse;
+import org.jboss.dmr.ModelNode;
+import org.jboss.threads.AsyncFuture;
+
+import javax.net.ssl.SSLContext;
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+
+/**
+ * Controller provides execution of {@link ModelNode} or {@link Operation} over HTTP.
+ * Asynchronous execution is not provided. Controller does not inform about any progress, therefore, there is no point
+ * in using {@link OperationMessageHandler}.
+ * Execution on {@link Operation} is allowed with <b>no attachments</b>.
+ */
+class HttpModelControllerClient implements ModelControllerClient {
+    private static final int NO_TIMEOUT = 0;
+
+    private final String url;
+    private final RequestConfig requestConfig;
+    private final String host;
+    private final int port;
+    private final String username;
+    private final String password;
+    private final String managementRealm;
+    private final Registry<ConnectionSocketFactory> registry;
+    private final SSLContext sslContext;
+    private final SSLConnectionSocketFactory sslConnectionSocketFactory;
+
+    private HttpModelControllerClient(
+            String scheme, String host, int port, String username, String password, int timeoutMillis, SslOptions ssl) {
+        this.host = host;
+        this.port = port;
+        this.username = username;
+        this.password = password;
+        url = String.format("%s://%s:%s/management", scheme, host, port);
+        RequestConfig.Builder configBuilder = RequestConfig.copy(RequestConfig.DEFAULT);
```

Comment:
Btw, `RequestConfig.copy(RequestConfig.DEFAULT)` seems to be completely the same as `RequestConfig.custom`.